pkg/querier/querier_test.go

```diff
@@ -1220,6 +1220,101 @@ func TestConfig_ValidateLimits(t *testing.T) {
+			// * originalT is the original query maxT, now + 2 hours
+			// * referenceT is now
+			// * since the query maxT can never be more than 1 hour into the future,
+			// the original maxT will be clamped backwards in time to now + 1 hour
+			testName:            "clamp maxT due to max query into future",
+			originalT:           now.Add(2 * time.Hour).UnixMilli(),
+			referenceT:          now.UnixMilli(),
+			limit:               1 * time.Hour,
+			clampMax:            true,
+			expectedClampedTime: now.Add(1 * time.Hour).UnixMilli(),
+		},
+		{
+			// scenario:
+			// * limit set to truncate the label query range if the query range is longer than 6 hours
+			// * originalT is the original query minT, now - 12 hours
+			// * referenceT is the query maxT, now - 3 hours
+			// * since the entire label query range cannot be longer than 6 hours
+			// the original minT will be clamped forwards in time to query maxT - 6 hours
+			testName:            "clamp minT due to max label query length",
+			originalT:           now.Add(-12 * time.Hour).UnixMilli(),
+			referenceT:          now.Add(-3 * time.Hour).UnixMilli(),
+			limit:               -6 * time.Hour,
+			clampMax:            false,
+			expectedClampedTime: now.Add(-9 * time.Hour).UnixMilli(),
+		},
+		{
+			// scenario:
+			// * limit set to only query the block store if the query range is previous to the last 4 hours
+			// * originalT is the original query maxT, now - 3 hours, sent to the block store querier
+			// * since the block store querier should only be queried for data older than the last 4 hours,
+			// the original maxT will be clamped backwards in time to now - 4 hours
+			testName:            "clamp maxT due to query store after",
+			originalT:           now.Add(-3 * time.Hour).UnixMilli(),
+			referenceT:          now.UnixMilli(),
+			limit:               -4 * time.Hour,
+			clampMax:            true,
+			expectedClampedTime: now.Add(-4 * time.Hour).UnixMilli(),
+		},
+		{
+			// scenario:
+			// * limit set to only query the ingesters if the query range is within the last 6 hours
+			// * originalT is the original query minT, now - 9 hours, sent to the ingester querier
+			// * since the ingester querier should only be queried for data within than the last 6 hours,
+			// the original minT will be clamped forwards in time to now - 6 hours
+			testName:            "clamp minT due to query ingesters within",
+			originalT:           now.Add(-9 * time.Hour).UnixMilli(),
+			referenceT:          now.UnixMilli(),
+			limit:               -6 * time.Hour,
+			clampMax:            false,
+			expectedClampedTime: now.Add(-6 * time.Hour).UnixMilli(),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.testName, func(t *testing.T) {
+			clampedTime := clampTime(
+				logger, test.originalT, test.referenceT, test.limit, "", test.clampMax,
+			)
+			assert.InDelta(t, test.expectedClampedTime, clampedTime, allowedDeltaT)
```

Comment:
Why is `InDelta()` required here? Might be good to add a comment explaining why if we can't use `Equals()`.

Reply:
Yeah, I'm not understanding why we needed a delta here. Usually, delta methods are for comparing floats since they are inexact. However, we dealing with durations and `in64` here so we should be able to assert an exact value.

Reply:
ah, there's a lot of InDelta used even on int64s in the querier tests so I did the same; but maybe it's to deal with time.Now() getting called at unpredictable times in the tests for larger paths.

Will update